fem/gslib.hpp

```diff
@@ -133,49 +135,106 @@ class FindPointsGSLIB
+   virtual const Array<unsigned int> &GetProc() const { return gsl_proc; }
    /// Return reference coordinates for each point found by FindPoints.
-   const Vector &GetReferencePosition() const { return gsl_mfem_ref;  }
+   virtual const Vector &GetReferencePosition() const { return gsl_mfem_ref;  }
    /// Return distance Distance between the sought and the found point
    /// in physical space, for each point found by FindPoints.
-   const Vector &GetDist()              const { return gsl_dist; }
+   virtual const Vector &GetDist()              const { return gsl_dist; }
 
    /// Return element number for each point found by FindPoints corresponding to
    /// GSLIB mesh. gsl_mfem_elem != gsl_elem for mesh with simplices.
-   const Array<unsigned int> &GetGSLIBElem() const { return gsl_elem; }
+   virtual const Array<unsigned int> &GetGSLIBElem() const { return gsl_elem; }
    /// Return reference coordinates in [-1,1] (internal range in GSLIB) for each
    /// point found by FindPoints.
-   const Vector &GetGSLIBReferencePosition() const { return gsl_ref; }
+   virtual const Vector &GetGSLIBReferencePosition() const { return gsl_ref; }
+};
+
+/** \brief OversetFindPointsGSLIB enables use of findpts for arbitrary number
+ * of overlapping grids. The parameters in this class are the same as
+ * FindPointsGSLIB with the difference of additional inputs required to account
+ * for more than 1 mesh.
+ */
+class OversetFindPointsGSLIB : public FindPointsGSLIB
+{
+protected:
+   bool overset;
+   unsigned int u_meshid;
+   Vector distfint; // Used to store the nodal values of the gridfunction passed
+   // to findpts.
+
+public:
+   OversetFindPointsGSLIB() : FindPointsGSLIB(),
+      overset(true) { }
+
+#ifdef MFEM_USE_MPI
+   OversetFindPointsGSLIB(MPI_Comm _comm) : FindPointsGSLIB(_comm),
+      overset(true) { }
+#endif
+
+   /** Initializes the internal mesh in gslib, by sending the positions of the
+       Gauss-Lobatto nodes of the input Mesh object @a m.
+       Note: not tested with periodic (DG meshes).
+       Note: the input mesh @a m must have Nodes set.
+
+       @param[in] m         Input mesh.
+       @param[in] meshid    A unique # for each overlapping mesh. This id is
+                            used to make sure that points being searched are not
+                            looked for in the mesh that they belong to.
+       @param[in] gfmax     (Optional) GridFunction in H1 that is used as a
+                            discriminator where one point is located in multiple
+                            meshes. The mesh that maximizes gfmax is chosen.
+       @param[in] bb_t      Relative size of bounding box around each element.
+       @param[in] newt_tol  Newton tolerance for the gslib search methods.
+       @param[in] npt_max   Number of points for simultaneous iteration. This
+                            alters performance and memory footprint.*/
+   void Setup(Mesh &m, const int meshid, GridFunction *gfmax = NULL,
```

Comment:
clang gives `overloaded virtual function` warning for `void Setup` and for `void FindPoints`. You might want to add 
````cpp
   using FindPointsGSLIB::Setup;        
   using FindPointsGSLIB::FindPoints;     
````